lib/datadog/tracing/contrib/excon/configuration/settings.rb

```diff
@@ -39,20 +41,11 @@ class Settings < Contrib::Configuration::Settings
             option :error_status_codes do |o|
               o.env Ext::ENV_ERROR_STATUS_CODES
               o.default 400...600
-              o.env_parser do |value|
-                values = if value.include?(',')
-                           value.split(',')
-                         else
-                           value.split
-                         end
-                values.map! do |v|
-                  v.gsub!(/\A[\s,]*|[\s,]*\Z/, '')
-
-                  v.empty? ? nil : v
-                end
-
-                values.compact!
-                values
+              o.setter do |v|
+                Tracing::Contrib::StatusRangeMatcher.new(v) if v
```

Comment:
I would add a runtime type check here, to enforce that either one of these is true:

- v is an `Integer`
- v is a `Range` with `.first` (and last, but that's internally checked by `Range`) being `Integer`